src/endpoint/sts/ops/sts_post_assume_role.js

```diff
@@ -0,0 +1,51 @@
+/* Copyright (C) 2016 NooBaa */
+'use strict';
+
+const dbg = require('../../../util/debug_module')(__filename);
+const { StsError } = require("../sts_errors");
+
+/**
+ * https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html
+ */
+async function assume_role(req) {
+    dbg.log0('sts_post_assume_role parmas: ', req.params);
+    let assumed_role;
+    try {
+        assumed_role = await req.sts_sdk.get_assumed_role(req);
+    } catch (err) {
+        if (err.code === 'ACCESS_DENIED') {
+            throw new StsError(StsError.AccessDeniedException);
+        }
+        throw new StsError(StsError.InternalFailure);
+    }
+    // Temporary credentials are NOT stored in noobaa
+    // TODO: need to generate session token and store in it 
+    // the temporary credentials and expiry
+    const access_keys = await req.sts_sdk.generate_temp_access_keys(req.params);
+    return {
+        AssumeRoleResponse: {
+            // https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumedRoleUser.html
+            AssumedRoleUser: {
+                Arn: `arn:aws:sts::${assumed_role.account_id}:assumed-role/${assumed_role.role_config.role_name}/${req.params.role_session_name}`,
+                AssumedRoleId: `${assumed_role.account_id}:${req.params.role_session_name}`
+            },
+            Credentials: {
+                AccessKeyId: access_keys.access_key.unwrap(),
+                SecretAccessKey: access_keys.secret_key.unwrap(),
+                Expiration: '',
+                SessionToken: ''
+            },
+            PackedPolicySize: 0
+        }
+    };
+}
+
+module.exports = {
+    handler: assume_role,
+    body: {
+        type: 'xml',
+    },
```

Comment:
does this request take any body at all?

Reply:
yes, ?Version=2011-06-15&Action=... being passed through the body

Reply:
I think that body type here should be 'x-www-form-urlencoded' and not 'xml' (although you currently ignore that value because you have just a single operation in sts_rest and you parse the body just once in the main handle_request function.

What I would do is move the parsing of such body from sts_rest to http_utils and invoke it from [read_and_parse_body](https://github.com/noobaa/noobaa-core/blob/64cbdbaf6721ba2538e2564b177f201bc34a25c5/src/util/http_utils.js#L228) when the body.type specifies it and the request content-type header agrees...

Reply:
I agree regarding the part that the type here should be 'x-www-form-urlencoded' and i'll move the parsing to read_and_parse_body..
but in order to identify the op (sts_post_assume_role) I need to read the body, so I didn't ignore the body.type because I have only one function, I did it because I had to read the body in order to identify the op

Reply:
I see so since `Action=AssumeRole` is inside the body you will have to read_and_parse_body just based on the content-type header, which will have to be called before resolving the op. Sounds ok.